myth/Themers/LocalizedViewThemer.php

```diff
@@ -0,0 +1,1165 @@
+     * be anything, but by default are used to render specific templates
+     * for desktop, tablet, and phone. The name of the variant is added
+     * to the view name, joined by a "+" symbol.
+     *
+     * Example:
+     *      $this->setVariant('phone');
+     *      $this->display('header');
+     *
+     *      Tries to display "views/header+phone.php"
+     *
+     * @param $variant
+     * @return $this
+     */
+    public function setVariant($variant)
+    {
+        if (isset($this->variants[$variant])) {
+            $this->current_variant = $variant;
+        }
+
+        return $this;
+    }
+    //--------------------------------------------------------------------
+
+    /**
+     * Adds a new variant to the system.
+     *
+     * @param $name
+     * @param $postfix
+     * @return $this|mixed
+     */
+    public function addVariant($name, $postfix)
+    {
+        $this->variants[$name] = $postfix;
+
+        return $this;
+    }
+
+    //--------------------------------------------------------------------
+
+    /**
+     * Removes a variant from the system.
+     *
+     * @param $name
+     * @return $this|mixed
+     */
+    public function removeVariant($name)
+    {
+        if (isset($this->variants[$name])) {
+            unset($this->variants[$name]);
+        }
+
+        return $this;
+    }
+
+    //--------------------------------------------------------------------
+    // Private Methods
+    //--------------------------------------------------------------------
+
+    /**
+     * Handles the actual loading of a view file, and checks for any
+     * overrides in themes, etc.
+     *
+     * @param $view
+     * @param $data
+     *
+     * @return string
+     */
+    private function loadView($view, $data)
+    {
+        // First - does it exist in the current theme?
+        $theme = ! empty($this->active_theme) ? $this->active_theme : $this->default_theme;
+        $theme = ! empty($this->folders[$theme]) ? $this->folders[$theme] : $theme;
+        $theme = rtrim($theme, '/ ') .'/';
+		
+		// Set localization idioms
+		$theme_lang_idom = str_replace(FCPATH .'themes/', FCPATH .'themes/'. CURRENT_LANGUAGE .'/', $theme);
+		$view_lang_idom = str_replace(FCPATH .'themes/', FCPATH .'themes/'. CURRENT_LANGUAGE .'/', $view);
+
+		if(defined('CURRENT_LANGUAGE') && file_exists($theme_lang_idom ."{$view}.php"))
+		{
+			$output = $this->ci->load->view_path( $theme_lang_idom . $view, $data, TRUE );
+		}
+		
+		elseif (file_exists($theme ."{$view}.php"))
+        {
+            $output = $this->ci->load->view_path( $theme . $view, $data, TRUE );
+        }
+		
+		elseif(defined('CURRENT_LANGUAGE') && realpath($view_lang_idom .".php"))
+		{
+			$output = $this->ci->load->view_path( $view_lang_idom, $data, TRUE );
+		}
+		
+        // Next, if it's a real file with path, then load it
+        elseif ( realpath( $view . '.php' ) )
+        {
+            $output = $this->ci->load->view_path( $view, $data, TRUE );
```

Comment:
But user might have modules and want them to have i18n support so views in modules should use idom too... or am i wrong?